ntbs-service/Pages/Notifications/Edit/Table/ManualTestResult.cshtml.cs

```diff
@@ -47,27 +47,49 @@ protected override async Task ValidateAndSave()
 
             if (TryValidateModel(TestResultForEdit, "TestResultForEdit"))
             {
-                if (RowId != null)
+                if (RowId == null)
                 {
                     await _testResultsRepository.AddTestResultAsync(TestResultForEdit);
                 }
                 else
                 {
-                    await _testResultsRepository.UpdateTestResultAsync(TestResultForEdit);
+                    TestResultForEdit.ManualTestResultId = RowId.Value;
+                    await _testResultsRepository.UpdateTestResultAsync(Notification, TestResultForEdit);
                 }
             }
         }
 
+        public async Task<IActionResult> OnPostDeleteAsync()
+        {
+            Notification = await GetNotificationAsync(NotificationId);
+            if (!(await AuthorizationService.CanEdit(User, Notification)))
+            {
+                return ForbiddenResult();
```

Comment:
Oh this makes me think that we need to add a 403 page after all which is identical to the access denied page - because our new error handler will be looking for this when ForbiddenResult is thrown!

Reply:
yes, i think it'd be good to still have it available for situation when we do want to throw it! :)